pkg/ottl/ottlfuncs/func_replace_all_patterns_test.go

```diff
@@ -60,7 +60,67 @@ func Test_replaceAllPatterns(t *testing.T) {
 			function: optionalArg,
 			want: func(expectedMap pcommon.Map) {
 				expectedMap.PutStr("test", "4804d6b7f03268e33f78c484977f3d81771220df07cc6aac4ad4868102141fad world")
-				expectedMap.PutStr("test2", "4804d6b7f03268e33f78c484977f3d81771220df07cc6aac4ad4868102141fad")
+				expectedMap.PutStr("test2", "09648f12e7a3940f539bb65d32321c2f96aa94ef698d87816e94d822c6f9d7c4")
```

Comment:
I'm sorry, I should have thought to mention this in the original PR, and I realize it's also late for this PR, but can we change the replacement function to be something other than a hash for the tests? Right now it's difficult to quickly independently verify that this is working correctly. I think a testing-only function that wraps the match in brackets or something could be a good option.

Reply:
I used this online tool to validate the hash values - https://emn178.github.io/online-tools/sha256.html. So one can enter the replacement string in the tool and quickly validate the hash value. Just wanted to call this out, I could start looking into wrapper function for tests for this [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27820) soon, please let me know if you'd like to see this test in this PR.

Reply:
I appreciate you providing that link and verifying the output for the tests, but if it's not too much trouble, I think it would be an appropriate change for this PR. I really want to make sure that it's easy to tell from the tests themselves that the output is correct.

It should be fairly easy to copy the `SHA256` factory definition into this test file and change it to the necessary function, I think most of the changes will end up in the test output.

Reply:
Added a test replacement function and modified the replace editor tests in this [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29408/commits/da61c1fde86b4f69b3a2c4b0f8c6d5622b0ed80f)

Reply:
@evan-bradley I've removed all hashes from tests in the last commit and replaced it with a test replacement function for quick validation as discussed here. The replacement function replaces a `string` with `hash(string)` to demonstrate hashing.